torch/jit/_recursive.py

```diff
@@ -518,6 +518,10 @@ def check_module_initialized(mod):
     if not hasattr(mod, '_parameters'):
         raise RuntimeError("'{}' has not been initialized, did you forget to call 'super()'?"
                            .format(torch.typename(type(mod))))
+    for param in mod.parameters():
+        if isinstance(param, torch.nn.parameter.UninitializedParameter):
+            raise RuntimeError("'{}' has uninitialized parameters or buffers, did you forget to run a forward pass"
```

Comment:
This needs more debugging ad it might not be needed

Reply:
What's the plan to check this? Shall we ask someone from the jit side to take a look?

Reply:
cc @suo

Reply:
would be nice 😅

Reply:
This seems fine—from what I understand, after the first forward we'll have fully initialized the parameters and the `LazyModule` will be indistinguishable from a regular `nn.Module`, right? If so, then it's fine to just error and ask the user to run forward.